app/src/js/reducers/api.js

```diff
@@ -27,40 +26,38 @@ export const initialState = {
 };
 
 export default createReducer(initialState, {
-  [DELETE_TOKEN]: (state) => {
-    set(state, 'tokens.token', null);
+  [DELETE_TOKEN]: ({ tokens }) => {
     setToken('');
+    tokens.token = null;
   },
-  [LOGIN]: (state) => {
-    set(state, 'authenticated', true);
-    set(state, 'inflight', false);
+  [LOGIN]: (draftState) => {
+    draftState.authenticated = true;
+    draftState.inflight = false;
   },
-  [LOGIN_INFLIGHT]: (state) => {
-    set(state, 'inflight', true);
+  [LOGIN_ERROR]: (draftState, { error }) => {
+    draftState.error = error;
+    draftState.inflight = false;
+    draftState.authenticated = false;
   },
-  [LOGIN_ERROR]: (state, action) => {
-    set(state, 'error', action.error);
-    set(state, 'inflight', false);
-    set(state, 'authenticated', false);
+  [LOGIN_INFLIGHT]: (draftState) => {
+    draftState.inflight = true;
   },
-  [LOGOUT]: (state) => {
-    set(state, 'authenticated', false);
+  [LOGOUT]: (draftState) => {
+    draftState.authenticated = false;
   },
-  [REFRESH_TOKEN]: (state, action) => {
-    set(state, 'tokens.error', null);
-    set(state, 'tokens.inflight', false);
-    set(state, 'tokens.token', action.token);
-    setToken(action.token);
+  [REFRESH_TOKEN]: ({ tokens }, { token }) => {
+    setToken(tokens.token = token);
```

Comment:
Is this clearer than in two steps?  Because I have to go look up what it means.

Reply:
I would very much prefer a line that sets draftstate.tokens.token = token, and another that sets the window store's token value.

Reply:
Agreed. Will do.